File: 0011-container-with-most-water/0011-container-with-most-water.c
int maxArea (int* height, int heightSize) {
    size_t r = 0;
    for (size_t i = 0, j = heightSize - 1; i != j; ) {
        r = fmax (r, fmin (height [i], height [j]) * (j - i));
        height [i] < height [j] ? ++i : --j;
    }
    return r;
}

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea (int [] height) {
        int r = 0;
        for (int i = 0, j = height.length - 1; i != j; ) {
            r = Math.max (r, Math.min (height [i], height [j]) * (j - i));
            if (height [i] < height [j])  ++i; else --j;
        }
        return r;
    }
}